Source/sound.cpp

```diff
@@ -382,44 +368,32 @@ void __fastcall sound_create_primary_buffer(int music_track)
 
 int __fastcall sound_DirectSoundCreate(GUID *guid, IDirectSound **DS, int always_null)
 {
-	IDirectSound **v3; // ebp
-	int v4; // eax
-	FARPROC v5; // ebx
-	int v6; // eax
-	GUID *v8; // [esp+10h] [ebp-4h]
-
-	v3 = DS;
-	v8 = guid;
-	if ( !hDsound_dll )
-	{
+	if ( !hDsound_dll ) {
 		hDsound_dll = LoadLibrary("dsound.dll");
-		if ( !hDsound_dll )
-		{
-			v4 = GetLastError();
-			ErrDlg(IDD_DIALOG5, v4, "C:\\Src\\Diablo\\Source\\SOUND.CPP", 422);
+		if ( !hDsound_dll ) {
+			ErrDlg(IDD_DIALOG5, GetLastError(), "C:\\Src\\Diablo\\Source\\SOUND.CPP", 422);
 		}
 	}
-	v5 = GetProcAddress(hDsound_dll, "DirectSoundCreate");
-	if ( !v5 )
-	{
-		v6 = GetLastError();
-		ErrDlg(IDD_DIALOG5, v6, "C:\\Src\\Diablo\\Source\\SOUND.CPP", 427);
+
+	FARPROC proc = GetProcAddress(hDsound_dll, "DirectSoundCreate");

```

Comment:
proc should be properly declared to match the function definition of DirectSoundCreate and the cast below removed.

Reply:
I have done my best to fix this now, but I'm not sure if it's as should. Might it work to include dsound.h and then have access to function without this or would it not be bin exact?